code/modules/mob/living/simple_animal/hostile/goose.dm

```diff
@@ -36,13 +36,30 @@
 	var/icon_vomit = "vomit"
 	var/icon_vomit_end = "vomit_end"
 	var/message_cooldown = 0
+	var/choking = FALSE
 
-/mob/living/simple_animal/hostile/retaliate/goose/handle_automated_movement()
+/mob/living/simple_animal/hostile/retaliate/goose/Initialize()
 	. = ..()
-	if (stat == DEAD)
+	RegisterSignal(src, COMSIG_MOVABLE_MOVED, .proc/goosement)
+
+/mob/living/simple_animal/hostile/retaliate/goose/proc/goosement(atom/movable/AM, OldLoc, Dir, Forced)
+	if(stat == DEAD)
 		return
-	if(prob(5) && random_retaliate == TRUE)
-		Retaliate()
+	for(var/obj/item/tasty in get_turf(src))
+		feed(tasty)
```

Comment:
This will make it eat everything near instantly, and you can conceivably get pretty large stacks of stuff to iterate through.
Look into slowing this down, both to make the goose eating stuff take longer and reduce impact on server/client performance.